markdown_help: Change "a" tag to "span" tag for spoiler-button class.

nikhilmaske-2001 opened the pull request:
Issue: The spoiler button present in the `Keyboard shortcut > Message formatting` page does not work. Related [thread](https://chat.zulip.org/#narrow/stream/9-issues/topic/Spoiler.20button.20not.20working).

Testing plan: Tested manually.

GIFs or screenshots:
Before:
![spoiler-button before](https://user-images.githubusercontent.com/59444243/105014620-a5d07c00-5a66-11eb-86f4-a806f3fd7ac9.gif)

After:
![spoiler-button after](https://user-images.githubusercontent.com/59444243/105014643-aec14d80-5a66-11eb-9ecf-961109f5e23b.gif)

amanagr commented:
Instead of writing the change you did in the code in the commit message, can you write why the change was made and how this is a fix for the issue.

timabbott commented:
Other than the comment about the commit message, this lgtm. 

(Though we should hopefully be soon replacing this infrastructure via #16891, though that seems to be abandoned -- want to do a PR based on that one that addresses the feedback there?)

nikhilmaske-2001 commented:
@amanagr I have updated the 

> Instead of writing the change you did in the code in the commit message, can you write why the change was made and how this is a fix for the issue.

@amanagr I have updated the commit, please check it out.

nikhilmaske-2001 commented:
> Other than the comment about the commit message, this lgtm.
> 
> (Though we should hopefully be soon replacing this infrastructure via #16891, though that seems to be abandoned -- want to do a PR based on that one that addresses the feedback there?)

@timabbott Yes I want to, can I continue the work in this same PR or make a new PR?

timabbott commented:
Merged, thanks @nikhilmaske-2001!  Please do a new PR for that cleanup.
